update/common.go

```diff
@@ -0,0 +1,124 @@
+package update
+
+import (
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+)
+
+// TxInfo defines the structure which hold the tx info
+type TxInfo struct {
+	TxHash []byte
+	Tx     data.TransactionHandler
+}
+
+// GetPendingMiniBlocks get all the pending miniBlocks from epoch start metaBlock and unFinished metaBlocks
+func GetPendingMiniBlocks(
+	epochStartMetaBlock *block.MetaBlock,
+	unFinishedMetaBlocksMap map[string]*block.MetaBlock,
+) ([]block.MiniBlockHeader, error) {
+
+	if epochStartMetaBlock == nil {
+		return nil, ErrNilEpochStartMetaBlock
+	}
+	if unFinishedMetaBlocksMap == nil {
+		return nil, ErrNilUnFinishedMetaBlocksMap
+	}
+
+	pendingMiniBlocks := make([]block.MiniBlockHeader, 0)
+	nonceToHashMap := CreateNonceToHashMap(unFinishedMetaBlocksMap)
+
+	for _, shardData := range epochStartMetaBlock.EpochStart.LastFinalizedHeaders {
+		computedPending, err := ComputePendingMiniBlocksFromUnFinished(
+			shardData,
+			unFinishedMetaBlocksMap,
+			nonceToHashMap,
+			epochStartMetaBlock.GetNonce(),
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		pendingMiniBlocks = append(pendingMiniBlocks, computedPending...)
+	}
+
+	//TODO: These pendingMiniBlocks are already added in method ComputePendingMiniBlocksFromUnFinished called above
+	//when nonce == epochStartMetaBlockNonce in "for" done there, by calling method GetAllMiniBlocksWithDst inside it.
+	//The below commented code should be removed, otherwise it would duplicate these miniblocks
+	//(especially the rewards miniblocks) which would be executed twice!
+	//for _, mbHdr := range epochStartMetaBlock.MiniBlockHeaders {
+	//	if mbHdr.SenderShardID != core.MetachainShardId || mbHdr.Type == block.PeerBlock {
+	//		continue
+	//	}
+	//	pendingMiniBlocks = append(pendingMiniBlocks, mbHdr)
+	//}
+
+	return pendingMiniBlocks, nil
+}
+
+// CreateNonceToHashMap creates a map of nonce to hash from all the given metaBlocks
+func CreateNonceToHashMap(unFinishedMetaBlocks map[string]*block.MetaBlock) map[uint64]string {
```

Comment:
do we need all these functions exported?

Reply:
No, only GetPendingMiniBlocks, I will make them unexported